app/src/androidTest/kotlin/org/wordpress/aztec/demo/tests/MixedTextFormattingTests.kt

```diff
@@ -108,4 +108,84 @@ class MixedTextFormattingTests : BaseTest() {
                 .toggleHtml()
                 .verifyHTML(text)
     }
+
+    @Test
+    fun testTwoHeadings() {
+        val text = "some text"
+        val linebreak = "\n"
+        val html = "<h1>$text</h1>$linebreak<h2>$text</h2>"
+
+        EditorPage()
+                .makeHeader(EditorPage.HeadingStyle.ONE)
+                .insertText(text)
+                .insertText(linebreak)
+                .makeHeader(EditorPage.HeadingStyle.TWO)
+                .insertText(text)
+                .toggleHtml()
+                .verifyHTML(html)
+    }
+
+    @Test
+    fun testEndHeadingFormatting() {
+        val text = "some text"
+        val linebreak = "\n"
+        val html = "<h1>$text</h1>$linebreak$text"
+
+        EditorPage()
+                .makeHeader(EditorPage.HeadingStyle.ONE)
+                .insertText(text)
+                .insertText(linebreak)
+                .insertText(text)
+                .toggleHtml()
+                .verifyHTML(html)
+    }
+
+    @Test
+    fun testEndQuoteFormatting() {
+        val text = "some text"
+        val linebreak = "\n"
+        val html = "<blockquote>$text</blockquote>$linebreak$text"
+
+        EditorPage()
+                .toggleQuote()
+                .insertText(text)
+                .insertText(linebreak)
+                .insertText(linebreak)
+                .insertText(text)
+                .toggleHtml()
+                .verifyHTML(html)
+    }
+
+    @Test
+    fun testRemoveQuoteFormatting() {
+        val text = "some text"
+        val linebreak = "\n"
+        val html = "<blockquote>$text</blockquote>$linebreak$text"
+
+        EditorPage()
+                .toggleQuote()
+                .insertText(text)
+                .insertText(linebreak)
+                .insertText(text)
+                .toggleQuote()
+                .toggleHtml()
+                .verifyHTML(html)
+    }
+
+    @Test
+    fun testQuotedListFormatting() {
+        var text = "some text\n"
+        val regex = Regex("<blockquote>\\s+<ul>[\\S\\s]+</ul>\\s+</blockquote>")
```

Comment:
I'm not aware how Regex is working in Kotlin/Java but I've able to make it work after removing escaped `\`: 
`<blockquote>\s+<ul>[\S\s]+<\/ul>\s+<\/blockquote>`

was matched:
```
<blockquote>
  <ul>
    <li>Wee</li>
    <li>Add</li>
    <li>Xox</li>
  </ul>
</blockquote>
```

Tested here: https://regexr.com/

Reply:
The backslashes need to be escaped per the [pattern syntax](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html). I'm not sure why the escaped version wouldn't work for you (although I'm also looking into stripping whitespace characters between HTML tags in our matchers for these tests, which will allow us to remove some of the regex here).